File: 2003/IntroToVBNETPart2/Presentation/EventTest1.cs
using System ;

public class MyEventClass
{
    // Declare a delegate
    public delegate void MyEventEventHandler ( int EventNumber ) ;
    // Declare an event for this class.
    public event MyEventEventHandler MyEvent  ;
    // Define a method that raises an event.
    public void CauseEvent( int EventNumber )
    {
        MyEvent ( EventNumber ) ;        // VB.NET: RaiseEvent MyEvent(EventNumber) ;
    }
}

public class TestEvent
{
    static void EventHandler( int EventNumber )
    {
        Console.WriteLine("Received event number {0}", EventNumber ) ;
    }

    public static void Main ( )
    {
        MyEventClass MyObject = new MyEventClass ( ) ;
        // AddHandler MyObject.MyEvent, AddressOf TestEvent.EventHandler 
        MyObject.MyEvent += new MyEventClass.MyEventEventHandler ( TestEvent.EventHandler ) ;
        // Cause the object to raise an event.
        MyObject.CauseEvent(123) ;
    }
}

File: 2003/IntroToVBNETPart2/Presentation/EventTest1.vb
Imports System

Public Class MyEventClass
    ' Declare an event for this class.
    Public Event MyEvent(ByVal EventNumber As Integer)
    ' Define a method that raises an event.
    Sub CauseEvent(ByVal EventNumber As Integer)
        RaiseEvent MyEvent(EventNumber)
    End Sub
End Class

Public Class TestEvent
    Shared Sub EventHandler(ByVal EventNumber As Integer)
        Console.WriteLine("Received event number {0}", EventNumber )
    End Sub

    Shared Sub Main()
        Dim MyObject As New MyEventClass()
        AddHandler MyObject.MyEvent, AddressOf TestEvent.EventHandler
        ' Cause the object to raise an event.
        MyObject.CauseEvent(123)
    End Sub
End Class
